docs/devGuide/writingComponents.md

```diff
@@ -0,0 +1,148 @@
+
+##### Attributes
+
+MarkBind attributes are passed to the Vue component as **props**. The type of the prop will be a `String`.
+
+##### Slots
+
+MarkBind slots are passed as **named slots** to the Vue component. The name of the MarkBind slot will be the same as the name of the Vue slot.
+Hence, MarkBind slots can be accessed in a Vue component either through the [named slots](https://v2.vuejs.org/v2/guide/components-slots.html#Named-Slots) or through the [`$slots` API](https://v2.vuejs.org/v2/api/#vm-slots).
+</panel>
+
+<br>
+
+{{ icon_examples }} 
+* As a wrapper for an external library (Modal component)
+* To implement a set of customised behaviours (Quiz component)
+
+### As a Plugin
+
+MarkBind components can be implemented as a plugin as well. 
+This is suitable for more lightweight components where the implementation is largely in processing the node, where it is fitting to use MarkBind Plugins' `processNode` or `postRender` interfaces.
+
+{{ icon_examples }} 
+* The `tree` component is implemented as a default plugin
+
+## Testing Components
+
+Automated tests that are relevant to the components include:
+
+* [Functional tests]({{baseUrl}}/devGuide/workflow.html#adding-test-site-content)
+* [Snapshot tests]({{baseUrl}}/devGuide/workflow.html#adding-snapshot-tests-for-components)
+
+The API for Snapshot tests can be found at [Vue Test Utils](https://v1.test-utils.vuejs.org/).
+
+Additionally, it's a good idea to check the deployed PR preview in addition to serving the app locally, to ensure that there are no differences.
+
+## Additional Considerations
+
+Some things you may need to consider when implementing a MarkBind component:
+
+#### Reactivity
+
+_Reactivity_ refers to the ability of a web framework to update your view whenever the application state has changed. 
+It is important to consider reactivity when implementing a component that may have dynamic contents that readers can interact with (e.g. opening a panel, triggering a tooltip to show).
+
+#### SSR
+
+Components should be compatible with SSR. 
+Minimally, there should be no SSR issues (viewable from the browser console), though a lack of warnings does **not** mean that there are no SSR problems. 
+A guide on SSR for MarkBind can be found [here]({{baseUrl}}/devGuide/design/serverSideRendering.html). 
+
+Vue-component-specific tips for resolving SSR issues:
+* The `mount` and `beforeMount` lifecycle hooks will only be executed on the client, not the server
+* When using `v-if`, ensure that it will evaluate to the same value on both the client and server 
+* Take note of how the Vue component will be compiled, ensure that the HTML is correct and aligns on both client- and server- side
+* Conditionally render data when it has been fully loaded
+
+#### Bundle size
+
+When creating a new component, you may need to import a package or library to support some functionality. 
+Ideally, this should not increase MarkBind's bundle size too much.
```

Comment:
Maybe can provide an example of how the bundle size can be checked? I think [bundlephobia](https://bundlephobia.com/) is quite a good site to check the size of a package/library but feel free to put in other methods if there are better alternatives :)

Reply:
Thanks for the suggestion! I had no idea that existed :o